core/src/main/scala/ai/h2o/sparkling/backend/external/ExternalBackendConf.scala

```diff
@@ -145,8 +147,10 @@ trait ExternalBackendConf extends SharedBackendConf with Logging {
 
   def setKillOnUnhealthyClusterDisabled(): H2OConf = set(PROP_EXTERNAL_CLUSTER_KILL_ON_UNHEALTHY._1, value = false)
 
+  @DeprecatedMethod("""set("spark.yarn.principal", principal)""", "3.32")
```

Comment:
@jakubhava I will have to remove the deprecation and keep `kerberosPrincipal` and `kerberosKeytab`. Spark doesn't allow to specify a proxy user and a principal at the same time. In Spark with Livy, Livy authenticates the principal with keytab and  sets proxy user via `klist`.

Reply:
oki oki, thanks for the info! I guess it applies also for the PR to master right?

Reply:
yes, you are right.